rust/operator/src/command.rs

```diff
@@ -37,50 +39,107 @@ pub fn prepare_container_cmd_args(kafka: &KafkaCluster) -> String {
+            STACKABLE_TLS_CLIENT_DIR,
+            SSL_STORE_PASSWORD,
+        ));
+        // Pipe the command config into the kafka-broker-api-version.sh via --command-config
+        args.push("|".to_string());
+        // Build the kafka-broker-api-version.sh call
+        args.push(build_readiness_check_command(
+            kafka,
+            broker_role_service_name,
+            namespace,
+        ));
     } else {
-        args.push("-b".to_string());
-        args.push(format!("localhost:{}", CLIENT_PORT));
+        args.push(build_readiness_check_command(
+            kafka,
+            broker_role_service_name,
+            namespace,
+        ));
     }
 
-    args.push("-L".to_string());
     args
 }
 
-fn kcat_client_auth_ssl(cert_directory: &str) -> Vec<String> {
-    vec![
-        "-X".to_string(),
-        "security.protocol=SSL".to_string(),
-        "-X".to_string(),
-        format!("ssl.key.location={cert_directory}/tls.key"),
-        "-X".to_string(),
-        format!("ssl.certificate.location={cert_directory}/tls.crt"),
-        "-X".to_string(),
-        format!("ssl.ca.location={cert_directory}/ca.crt"),
-    ]
+/// Generates input for the `kafka-broker-api-versions.sh` argument --command-config to
+/// work with an TLS encrypted / authenticated cluster
+fn build_command_config(tls_dir: &str, store_password: &str) -> String {
+    format!(
+        "printf \"security.protocol=SSL\n \
+             ssl.truststore.location={tls_dir}/truststore.p12\n \
+             ssl.truststore.password={store_password}\n \
+             ssl.keystore.location={tls_dir}/keystore.p12\n \
+             ssl.keystore.password={store_password}\""
+    )
 }
 
-fn kcat_client_ssl(cert_directory: &str) -> Vec<String> {
-    vec![
-        "-X".to_string(),
-        "security.protocol=SSL".to_string(),
-        "-X".to_string(),
-        format!("ssl.ca.location={cert_directory}/ca.crt"),
-    ]
+/// Generates the command and parameters for the `kafka-broker-api-versions.sh` to retrieve
+/// registered brokers
+fn build_readiness_check_command(
+    kafka: &KafkaCluster,
+    broker_role_service_name: &str,
+    namespace: &str,
+) -> String {
+    let port = kafka.client_port();
+
+    let mut command = vec![
+        "/stackable/kafka/bin/kafka-broker-api-versions.sh".to_string(),
+        "--bootstrap-server".to_string(),
+        format!("$POD_NAME.{broker_role_service_name}.{namespace}.svc.cluster.local:{port}"),
+    ];
+
+    // if TLS is activated, need the output from build_command_config() for TLS settings
+    if kafka.client_authentication_class().is_some() || kafka.client_tls_secret_class().is_some() {
+        command.extend(["--command-config".to_string(), "/dev/stdin".to_string()]);
+    }
+
+    // -c to count the occurrences of id (which reflects the amount of brokers)
+    command.extend([
+        "|".to_string(),
+        "grep".to_string(),
+        "-c".to_string(),
+        "id".to_string(),
+    ]);
```

Comment:
This is another option of writing.
We do we count the number of brokers here? I think this information is never used anywhere, is it? Would it make more sense to assert that at least one entry is in here? (I think we have to try out different constellations and see what this script returns)
```suggestion
    command.push("| grep -c id".to_string());
```

Reply:
yeah i think a test for >= 1 would be ok here?

Reply:
This works (a counter-test indicates as much), although something a little more declarative might be:
```
bash-4.4$ bin/kafka-broker-api-versions.sh --bootstrap-server simple-kafka-broker-default.kuttl-test-guiding-gnat.svc.cluster.local:9092 | awk '/^[0-9]/ {print $1}'
172.18.0.4:30528
bash-4.4$ 
```
so at least we log out the address. There aren't many options with that script.

Reply:
I like Andrew's solution. My guess would be that asserting on >= 1 is needless because we need a broker (localhost) to return us this answer but I'm not enough into this script.

Reply:
I basically want to fail if #brokers < 1 (probably wont happen but you never know)

Reply:
Sounds fine to me